File: 2486-most-frequent-even-element/most-frequent-even-element.java
class Solution {
    public int mostFrequentEven(int[] nums) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i:nums){
            if(i%2 == 0)map.put(i,map.getOrDefault(i,0)+1);
        }
        if(map.size() == 0) return -1;
        int max = 0;
        for(int value:map.values()){
            max = Math.max(max,value);
        }
        List<Integer> list = new ArrayList<>();
        for(int key:map.keySet()){
            if(map.get(key) == max)list.add(key);
        }
        Collections.sort(list);
        return list.get(0);
    }
}

File: 2486-most-frequent-even-element/most-frequent-even-element.cpp
class Solution {
public:
    int mostFrequentEven(vector<int>& nums) {
        unordered_map<int,int> map;
        set<int> set;
        for(int i:nums){
            if(i%2 == 0){
                map[i]++;
            }
        }
        if(map.size() == 0)return -1;
        int max = 0;
        for(const auto&pair:map){
            if(pair.second > max){
                max = pair.second;
            }
        }
        for(const auto & pair : map){
            if(map[pair.first] == max)set.insert(pair.first);
        }
        for(int i:set){
            return i;
        }
        return 0;
    }
};